cli/azd/pkg/experimentation/assignment.go

```diff
@@ -0,0 +1,186 @@
+package experimentation
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/azure/azure-dev/cli/azd/internal/tracing/resource"
+	"github.com/azure/azure-dev/cli/azd/pkg/config"
+	"github.com/azure/azure-dev/cli/azd/pkg/osutil"
+)
+
+// cCacheFileNamePattern is the pattern used to generate the cache file name. It is formatted
+// using the machine ID.
+const cCacheFileNamePattern = "mid-%s.cache"
+
+// MachineIdParameterName is the name of the parameter used to identify the machine ID in the assignment
+// request.
+const MachineIdParameterName = "machineid"
+
+// AssignmentsManager manages interaction with the Assignments service, caching the results for 24 hours.
+type AssignmentsManager struct {
+	cacheRoot string
+	client    *tasClient
+}
+
+// NewAssignmentsManager creates a new AssignmentManager, which will communicate with the TAS service. The
+// AssignmentManager caches the assignment information for 24 hours.
+
+func NewAssignmentsManager(endpoint string, transport policy.Transporter) (*AssignmentsManager, error) {
+	configRoot, err := config.GetUserConfigDir()
+	if err != nil {
+		return nil, err
+	}
+
+	cacheRoot := filepath.Join(configRoot, "experiments")
+	if err := os.MkdirAll(cacheRoot, osutil.PermissionDirectory); err != nil {
+		return nil, err
+	}
+
+	client, err := newTasClient(endpoint, &policy.ClientOptions{
+		Transport: transport,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return &AssignmentsManager{
+		cacheRoot: cacheRoot,
+		client:    client,
+	}, nil
+}
+
+// Assignment is a subset of the information returned by the TAS service.
```

Comment:
I'll be honest, I didn't put a ton of thought into the shapes here.  This is just the same as the wire format of the response, but without the JSON tags and the ImpressionID and FlightingVersion fields elided.  I think that we might want to provide some nicer APIs on top of the assignment struct once we know how we are going to use the assignment information inside `azd` to decide on if we should offer a choice or not.

Reply:
That makes sense, could make it a lot easier for consumption everywhere.